client/src/featureform/register.py

```diff
@@ -2572,6 +2572,50 @@ def apply(self):
         else:
             state().create_all(self._stub)
 
+    def wait(self, resource_type, name, variant=None, timeout=None):
+        # gets a resource and waits until status is set to ready
+        resource_functions = {
+            "user": self.get_user,
+            "model": self.get_model,
+            "entity": self.get_entity,
+            "provider": self.get_provider
+        }
+
+        resource_variant_functions = {
+            "feature": self.get_feature,
+            "label": self.get_label,
+            "source": self.get_source,
+            "trainingset": self.get_training_set,
+            "training-set": self.get_training_set,
+        }
+        
+        resource = None
+        if variant is None:
+            resource = resource_functions[resource_type](name, local=self.local)
+        else:
+            resource = resource_variant_functions[resource_type](name, variant, local=self.local)
+        status = resource.get_status()
+        
+        timeout_duration = timedelta(seconds=0)
```

Comment:
These timeouts could probably be returned from a function

Reply:
what do you mean?